task.py

```diff
@@ -275,7 +276,32 @@ def get_namespace(self) -> str:
     def get_duration(self) -> int:
         return self.ts.cfg_descr.get_tft().duration
 
+    def _fetch_default_resource_name(self) -> Union[str, None]:
```

Comment:
could we do:
```diff
-    def _fetch_default_resource_name(self) -> Union[str, None]:
-        data = self.client.oc_get("network-attachment-definition")
+    @staticmethod
+    @functools.cache
+    def _fetch_default_resource_name(client: K8sClient) -> Optional[str]:
+        data = client.oc_get("network-attachment-definition")
         resource_name = None
```

The purpose is to determine the value only once per run, instead of multiple times (and getting potentially different results). Also, each `client.oc_get()` will log data, which is good, but seeing it redundantly is confusing.